internal/provider/resource_workspace_current_state.go

```diff
@@ -0,0 +1,313 @@
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *workspaceCurrentStateResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_workspace_current_state"
+}
+
+func (t *workspaceCurrentStateResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	description := "Defines and manages a workspace current state."
+
+	resp.Schema = schema.Schema{
+		Version:             1,
+		MarkdownDescription: description,
+		Description:         description,
+		Attributes: map[string]schema.Attribute{
+			"workspace_path": schema.StringAttribute{
+				MarkdownDescription: "The full path of the workspace.",
+				Description:         "The full path of the workspace.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_source": schema.StringAttribute{
+				MarkdownDescription: "The source of the module, including the API hostname.",
+				Description:         "The source of the module, including the API hostname.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_version": schema.StringAttribute{
+				MarkdownDescription: "The version identifier of the module.",
+				Description:         "The version identifier of the module.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"teardown": schema.BoolAttribute{
+				MarkdownDescription: "Whether to teardown the deployment of the module to the workspace.",
+				Description:         "Whether to teardown the deployment of the module to the workspace.",
+				Optional:            true,
+				Computed:            true, // API sets a default value of false if not specified.
+				// Can be updated in place, so no RequiresReplace plan modifier.
+			},
+		},
+	}
+}
+
+// Configure lets the provider implement the ResourceWithConfigure interface.
+func (t *workspaceCurrentStateResource) Configure(_ context.Context,
+	req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	t.client = req.ProviderData.(*tharsis.Client)
+}
+
+func (t *workspaceCurrentStateResource) Create(ctx context.Context,
+	req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	// Retrieve values from workspace current state.
+	var workspaceCurrentState WorkspaceCurrentStateModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &workspaceCurrentState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If teardown is not specified, make it false.
```

Comment:
Create should only create a run if the current workspace state does not match the desired state.

Reply:
What determines desired state? If you provide the same module and version that has already been applied, it doesn't mean that the apply would be no-op. There could be other changes outside of the workspace that affect the outcome of the apply. We may need to treat everything as unknown and always apply. Otherwise the plan state should do the plan on the workspace to determine if there are changes regardless of module and version.

Reply:
If I understand correctly, if there are no changes needed in the target/secondary workspace, that plan stage will make that determination and that apply stage won't be triggered.